src/ucs/datastruct/mpool.c

```diff
@@ -56,37 +56,41 @@ static void ucs_mpool_chunk_leak_check(ucs_mpool_t *mp, ucs_mpool_chunk_t *chunk
     }
 }
 
-ucs_status_t ucs_mpool_init(ucs_mpool_t *mp, size_t priv_size,
-                            size_t elem_size, size_t align_offset, size_t alignment,
-                            unsigned elems_per_chunk, unsigned max_elems,
-                            ucs_mpool_ops_t *ops, const char *name)
+ucs_status_t ucs_mpool_init(ucs_mpool_params_t *params, ucs_mpool_t *mp)
 {
     /* Check input values */
-    if ((elem_size == 0) || (align_offset > elem_size) ||
-        (alignment == 0) || !ucs_is_pow2(alignment) ||
-        (elems_per_chunk == 0) || (max_elems < elems_per_chunk) ||
-        !ops || !ops->chunk_alloc || !ops->chunk_release)
+    if ((params->elem_size == 0) ||
+        (params->align_offset > params->elem_size) ||
+        (params->alignment == 0) || !ucs_is_pow2(params->alignment) ||
+        (params->elems_per_chunk == 0) ||
+        (params->max_elems < params->elems_per_chunk) ||
+        (params->ops == NULL) ||
+        (!params->ops->chunk_alloc || !params->ops->chunk_release) ||
+        (params->grow_factor < 1))
     {
         ucs_error("Invalid memory pool parameter(s)");
         return UCS_ERR_INVALID_PARAM;
     }
 
-    mp->data = ucs_malloc(sizeof(*mp->data) + priv_size, "mpool_data");
+    mp->data = ucs_malloc(sizeof(*mp->data) + params->priv_size, "mpool_data");
     if (mp->data == NULL) {
         ucs_error("Failed to allocate memory pool slow-path area");
         return UCS_ERR_NO_MEMORY;
     }
 
     mp->freelist              = NULL;
-    mp->data->elem_size       = sizeof(ucs_mpool_elem_t) + elem_size;
-    mp->data->alignment       = alignment;
-    mp->data->align_offset    = sizeof(ucs_mpool_elem_t) + align_offset;
-    mp->data->elems_per_chunk = elems_per_chunk;
-    mp->data->quota           = max_elems;
+    mp->data->elem_size       = sizeof(ucs_mpool_elem_t) + params->elem_size;
+    mp->data->grow_factor     = params->grow_factor;
+    mp->data->max_chunk_size  = params->grow_factor == 1 ?
+                                UINT_MAX : params->max_chunk_size;
```

Comment:
why need to check grow_factor? just take `params->max_chunk_size` as-is

Reply:
fixed.